docs/user/using-an-OIDC-provider.md

```diff
@@ -59,39 +61,80 @@ In the case of Google we can use an OAuth 2.0 client ID. You can find more infor
+
+**Example 1: Using the OIDC provider**
+
+This example uses `oauth2-proxy`'s generic OIDC provider with Google, but is applicable to any OIDC provider. Note that the issuer url is passed as an additional flag here, together with an option to enable the cookie being set over an insecure connection for local development only:
+
+```
+helm install bitnami/kubeapps \
+  --namespace kubeapps --name kubeapps \
+  --set authProxy.enabled=true \
+  --set authProxy.provider=oidc \
+  --set authProxy.clientID=my-client-id.apps.googleusercontent.com \
+  --set authProxy.clientSecret=my-client-secret \
+  --set authProxy.cookieSecret=$(echo "not-good-secret" | base64) \
+  --set authProxy.additionalFlags="{-cookie-secure=false,-oidc-issuer-url=https://accounts.google.com}" \
+```
+
+**Example 2: Using a custom provider**
+
+Some of the specific providers that come with `oauth2-proxy` are using OpenIDConnect to obtain the required IDToken and can be used instead of the generic oidc provider. Currently these include the GitLab, Google and LoginGov providers. The user authentication flow is the same as above, with some small UI differences, such as the default login button is customized to the provider (rather than "Login with OpenID Connect"), or improved presentation when accepting the requested scopes (as is the case with Google, but only visible if you request extra scopes).
 
+Here we no longer need to provide the issuer -url as an additional flag:
 ```
 helm install bitnami/kubeapps \
   --namespace kubeapps --name kubeapps \
   --set authProxy.enabled=true \
+  --set authProxy.provider=google \
   --set authProxy.clientID=my-client-id.apps.googleusercontent.com \
   --set authProxy.clientSecret=my-client-secret \
-  --set authProxy.discoveryURL=https://accounts.google.com
+  --set authProxy.cookieSecret=$(echo "not-good-secret" | base64) \
+  --set authProxy.additionalFlags="{-cookie-secure=false}"
+```
+
+**Example 3: Authentication for Kubeapps on a GKE cluster**
+
+Google Kubernetes Engine does not allow an OIDC IDToken to be used to authenticate requests to the managed API server, instead requiring the standard OAuth2 access token.
+For this reason, when deploying Kubeapps on GKE we need to ensure that
+
+* The scopes required by the user to interact with cloud platform are included, and
+* The Kubeapps frontend uses the OAuth2 `access_key` as the bearer token when communicating with the managed Kubernetes API
+
+Note that using the custom `google` provider here enables google to prompt the user for consent for the specific permissions requested in the scopes below, in a user-friendly way. You can also use the `oidc` provider but in this case the user is not prompted for the extra contsent:
```

Comment:
Typo: `contsent`

>  You can also use the `oidc` provider but in this case the user is not prompted for the extra contsent:

I couldn't follow.

Reply:
Thanks, fixed the typo. The extra consent is:

![google-signin-oauth2](https://user-images.githubusercontent.com/497518/66379021-66605680-ea00-11e9-8368-b4a17ea68174.png)

If you use the `oidc` provider, you don't get this (which is odd, but verified multiple times).